packages/js/src/dashboard/widgets/site-kit-setup-widget.js

```diff
@@ -211,28 +217,44 @@ const SiteKitSetupAction = ( { currentStep, config, isConnectionCompleted, onDis
  *
  * @param {DataProvider} dataProvider The data provider.
  * @param {RemoteDataProvider} remoteDataProvider The remote data provider.
+ * @param {DataTracker} dataTracker The data tracker.
  *
  * @returns {JSX.Element} The widget.
  */
-export const SiteKitSetupWidget = ( { dataProvider, remoteDataProvider } ) => {
+export const SiteKitSetupWidget = ( { dataProvider, remoteDataProvider, dataTracker } ) => {
+	const { grantConsent, dismissPermanently } = useSiteKitConfiguration( dataProvider, remoteDataProvider );
+	const currentStep = dataProvider.getSiteKitCurrentConnectionStep();
+	const config = dataProvider.getSiteKitConfiguration();
+	const isConnectionCompleted = dataProvider.isSiteKitConnectionCompleted() && config.isVersionSupported;
+
+	useEffect( () => {
+		const stepName = getCurrentStepName( currentStep );
+		if ( dataTracker.getSetupStepsTrackingElement( "setupWidgetLoaded" ) === "no" ) {
+			dataTracker.track( {
+				setupWidgetLoaded: "yes",
+				firstInteractionStage: stepName,
+				lastInteractionStage: stepName,
+			} );
+		} else if ( dataTracker.getSetupStepsTrackingElement( "setupWidgetLoaded" ) === "yes" ) {
+			dataTracker.track( { lastInteractionStage: stepName } );
+		}
+	}, [ dataTracker, currentStep ] );
```

Comment:
Lets move it to a hook :) We can move the the `getCurrentStepName` to that hook file.